File: python/integracion_con_bases_de_datos/guardar_multiples_contactos/guardar_multiples_contactos.py
from conexion import conectar
def guardar_contacto(nombre, telefono, email):
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        sql = "INSERT INTO contactos (nombre, telefono, email) VALUES (%s, %s, %s)"
        valores = (nombre, telefono, email)
        cursor.execute(sql, valores)
        conexion.commit()
        cursor.close()
        conexion.close()
        print("✅ Contacto guardado correctamente")
def main():
    print("=== Guardar Múltiples Contactos ===")
    while True:
        nombre = input("Nombre: ")
        telefono = input("Teléfono: ")
        email = input("Email: ")
        guardar_contacto(nombre, telefono, email)
        continuar = input("¿Deseas agregar otro contacto? (s/n): ").lower()
        if continuar != "s":
            print("📁 Fin del registro de contactos")
            break
if __name__ == "__main__":
    main()

File: java/integracion_con_base_de_datos/guardar_multiples_contactos/src/guardar_multiples_contactos/guardar_multiples_contactos.java
package guardar_multiples_contactos;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.util.Scanner;
public class guardar_multiples_contactos {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Connection conn = conexion.conectar();
        if (conn == null) {
            System.out.println("❌ No se pudo conectar");
            sc.close();
            return;
        }
        String opcion;
        do {
            System.out.println("\n=== NUEVO CONTACTO ===");
            System.out.print("Nombre: ");
            String nombre = sc.nextLine();
            System.out.print("Teléfono: ");
            String telefono = sc.nextLine();
            System.out.print("Correo: ");
            String correo = sc.nextLine();
            String sql = "INSERT INTO contactos (nombre, telefono, correo) VALUES (?, ?, ?)";
            try {
                PreparedStatement ps = conn.prepareStatement(sql);
                ps.setString(1, nombre);
                ps.setString(2, telefono);
                ps.setString(3, correo);
                ps.executeUpdate();
                System.out.println("✅ Contacto guardado");
            } catch (Exception e) {
                System.out.println("❌ Error: " + e.getMessage());
            }
            System.out.print("\n¿Desea agregar otro contacto? (s/n): ");
            opcion = sc.nextLine();
        } while (opcion.equalsIgnoreCase("s"));
        System.out.println("👋 Fin del programa");
        sc.close();
    }
}